### ISSUE TITLE
Git through SSH with passphrase-protected key whrn in WSL remote

### ISSUE THREAD

**Issue by ivangretsky** [eamodio/vscode-gitlens#909]
Good day! Thanks for a best-in-class extension!

I am trying to usr git through SSH in a WSL remote. The SSH connection is protected with a key pain which is password-protected.  How do I start an SSH agent and input that passphrase for a shell in which GitLens operates? 

By default a bach/wsl shell is opened in terminal window on WSL remote connection. I start an SSH agent there and add the key, but it seems like this is not the shell in which GitLens is sending its git commands.

TIA!

**Comment by clemjlt** [comment]
Hi ! Same situation here, how can we do that ?

Thanks,

**Comment by sebtiz13** [comment]
Hi ! Have same issue.
It's not possible to open a prompt to enter password for SSH Key ?

**Comment by mhmmd-ysf** [comment]
Maybe this could help
https://stackoverflow.com/a/60198798/11154427

TLDR:
- regenerate your key, make sure it's not passphrase-protected
- add the new public one in the ~/.ssh/authorized_keys file of the WSL session
- don't forget to clone via SSH not HTTPS

It worked for me

**Comment by ivangretsky** [comment]
> make sure it's not passphrase-protected
This line makes the answer invalid for the case)

**Comment by mhmmd-ysf** [comment]
Oops I didn't see that before, sorry 🤦🏻‍♂️

**Comment by eamodio** [comment]
This isn't something that I will likely add myself, but if the community wants to provide a PR I'm certainly open to it -- depending on the complexity and scope of the changes required.

**Comment by ivangretsky** [comment]
It seems like native git support does not allow this either, so probably nothing to fix in the extension. Created [an issue](https://github.com/microsoft/vscode-remote-release/issues/2369) closer to the source of the problem)

**Comment by Talmanes** [comment]
This works for me, but it can flake out from time to time.

1. From WSL2 terminal, using Windows Terminal Preview.
2. `eval $(ssh-agent -s)`
3. `ssh-add -k ~/.ssh/id_YOUR-KEY.GitHub`
4. Then start VSCode from this command line, `code project/` 

This is not 100%, there is likely some condition I am not paying attention to.  It might be a timeout or some other issue.  I just close VScode and try again.

**Comment by fai814** [comment]
Lennaert ***@***.***> 於 2021年11月2日週二 下午7:33寫道：
